src/Misc/tools.ts

```diff
@@ -912,7 +913,7 @@ export class Tools {
      * @return True if the uri is a base64 string or false otherwise
      */
     public static IsBase64(uri: string): boolean {
-        return uri.length < 5 ? false : uri.substr(0, 5) === "data:";
+        return uri.length < 5 ? false : uri.substr(0, 5) === "data:" && StringTools.EndsWith(uri.split(',')[0], ";base64");

```

Comment:
This is not efficient since split will create the second part of the string. Probably better to use `indexOf` to get the index of the `,` character and use substr to get the string before that index.

Reply:
Good call! I've updated the logic so it will only duplicate 7 characters of the string to validate that it's equal to ";base64"